src/features/transfer-ownership/TransferOwnershipForm/TransferOwnershipForm.test.tsx

```diff
@@ -0,0 +1,465 @@
+
+		// transaction in progress step
+		await screen.findByText('Your transaction is being processed...');
+
+		// assert successful request
+		await waitFor(() => {
+			expect(mockTranferDomainOwnership).toBeCalledTimes(1);
+			expect(mockTx).toBeCalledTimes(1);
+			expect(mockTranferDomainOwnership).toHaveBeenCalledWith(
+				mock.validInputAddress,
+				mock.domain.id,
+				{ isSigner: true },
+			);
+		});
+	});
+
+	describe('Details Step', () => {
+		test('should display correct domain data', async () => {
+			testSetup();
+
+			await screen.findByText(`0://${mock.domain.name}`);
+			await screen.findByText(mock.metadata.title);
+			await screen.findByText(truncateAddress(mock.domain.minter));
+		});
+
+		test('primary button should be disabled if input value is an empty string', async () => {
+			testSetup();
+
+			const onConfirmInputButton = await screen.findByRole('button', {
+				name: /transfer/i,
+			});
+
+			expect(onConfirmInputButton).toBeInTheDocument();
+			expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+
+			const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+			expect(input).toBeInTheDocument();
+
+			await act(async () => {
+				fireEvent.input(input, { target: { value: '' } });
+			});
+
+			expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+		});
+
+		test('should display error message when input value is not a valid eth address', async () => {
+			testSetup();
+
+			// input error message
+			const errorMessage = 'Please enter a valid Ethereum wallet address';
+
+			expect(screen.queryByText(errorMessage)).not.toBeInTheDocument();
+
+			const onConfirmInputButton = await screen.findByRole('button', {
+				name: /transfer/i,
+			});
+
+			expect(onConfirmInputButton).toBeInTheDocument();
+			expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+
+			const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+			expect(input).toBeInTheDocument();
+
+			await act(async () => {
+				fireEvent.input(input, {
+					target: { value: mock.inValidInputAddress },
+				});
+			});
+
+			expect(onConfirmInputButton).not.toHaveAttribute('aria-disabled');
+
+			fireEvent.click(onConfirmInputButton);
+
+			await screen.findByText(errorMessage);
+		});
+
+		test('should display error message when input value is equal to connected account address', async () => {
+			testSetup();
+
+			// input error message
+			const errorMessage = 'The address entered already owns this domain';
+
+			expect(screen.queryByText(errorMessage)).not.toBeInTheDocument();
+
+			const onConfirmInputButton = await screen.findByRole('button', {
+				name: /transfer/i,
+			});
+
+			expect(onConfirmInputButton).toBeInTheDocument();
+			expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+
+			const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+			expect(input).toBeInTheDocument();
```

Comment:
Can remove these `expect` as we validated them in previous tests - I'll stop commenting on this from here but yeah, you can cut down tests by reducing the scope of what they test and making sure not to double up on assertions.

Reply:
Done, reduced the scope on each test, looks much cleaner